feat: TournamentPage 토너먼트 대진표(2강) 및 기능 구현

yeni023 opened the pull request:
## 🔗 반영 브랜치
(#157) feature/Tournament-> dev

## 📝 작업 내용
- `MatchNode` 컴포넌트를 통한 재귀적 4강/2강 트리 렌더링
- `PlayerCard` 컴포넌트 구현 (기본/대형 카드, 승자 강조, 전적 팝업)
- `ready` 버튼 구현 및 클릭 시 사용자 테두리 표시
- 팝업 위치를 플레이어 위치에 따라 위/아래 조건부 렌더링
- `ChatBox`, `ChatInput` 컴포넌트를 통한 채팅 UI 완성

### 📸 스크린샷 (선택)
![image](https://github.com/user-attachments/assets/965df20c-775e-4772-8d9a-f4462279cc6e)

## 💬 리뷰 요구사항(선택)
- 현재 사용자 ID는 pong으로 하드코딩되어 있으며, 추후 API 연동 시 실제 사용자 정보로 대체할 예정입니다.
- 채팅 메시지 전송 기능은 구현되어 있으나, 보낸 사용자 정보 표시는 추후 백엔드 연동 시 적용할 계획입니다.
- 2강 대진표부터 우선 구현했으며, 4강 대진표 스타일 및 위치 조정 작업은 곧 이어서 진행하겠습니다.

yeni023 committed:
feat: TournamentPage 초기 구현

토너먼트 대진표 UI 및 기능 구현

- 4강/2강 구조의 재귀적 MatchNode 컴포넌트 구현
- PlayerCard 컴포넌트로 참가자 정보 및 승자 하이라이트 표시
- ready 버튼 추가 및 준비 상태에 따른 테두리 표시
- 마우스 호버 시 전적 팝업 표시
- 채팅 부분 UI 작업

HJ0311 commented on src/constants/routes.ts:
```diff
@@ -6,6 +6,7 @@ export const PATH = {
   HISTORY: '/history',
   GAME_SELECT: '/game/select',
+  TOURNAMENT: '/tournament',
 
```
game 관련 페이지는 /game/* 경로로 보내고 있어요! 게임 대기 방도 `/game/tournament` 이런식으로 넣어주시면 좋을 것 같습니다.

HJ0311 commented on src/pages/TournamentPage/components/Chat/ChatInput.tsx:
```diff
@@ -0,0 +1,44 @@
+
+  const handleSend = () => {
+    if (!input.trim()) return;
+    onSend(input.trim());
```
input.trim()을 두 번 호출하고 있는데 일관된 결과의 비교를 위해 한 번만 실행한 결과를 변수에 넣고 그 결과를 사용하는 게 더 좋을 것 같습니다!

HJ0311 commented on src/pages/TournamentPage/components/MatchNode/index.tsx:
```diff
@@ -0,0 +1,99 @@
+        <div className={styles.branchWrapper}>
+          <MatchNode match={left} side="left" readyIds={readyIds} />
+          <div className={styles.centerColumn}>
+            {isRoot ? (
```
root 아래의 내용이 너무 많아서 가독성이 떨어지는 것 같아요!
따로 함수로 분리 해 놓고 return 내부에는 어떤 내용이 들어가는구나에 대한 함수 이름만 들어가 있으면 좀 더 읽기 편하지 않을까 싶습니다.

HJ0311 commented on src/pages/TournamentPage/components/MatchNode/index.tsx:
```diff
@@ -0,0 +1,99 @@
+                    isReady={false}
+                  />
+                </div>
+                <MatchLines winner={winnerFromLeft === 'top' ? 'bottom' : 'top'} variant="right" />
```
삼항 연산자 따로 유틸 함수화 하면 더 보기 편할 것 같아요!

HJ0311 commented on src/pages/TournamentPage/components/PlayerCard/index.tsx:
```diff
@@ -0,0 +1,63 @@
+      onMouseLeave={() => setShowPopup(false)}
+    >
+      <div className={`${styles.avatarWrapper} ${borderClass}`}>
+        <img src={player.avatarUrl} className={avatarClass} />
```
alt 속성 추가하면 좋을 것 같습니다.

yeni023 committed:
Merge branch 'dev' into feature/Tournament

yeni023 committed:
refactor: MatchNode 컴포넌트 구조 분리 및 가독성 개선

- root/no-root 일 때의 중앙 매치 박스를 별도 함수로 분리
- 삼항 연산자 중첩 제거로 로직 명확화
- isReady 유틸 함수화하여 중복 제거 및 재사용성 향상

yeni023 committed:
fix: GameSelectPage 경로 삭제

hakilee commented on src/pages/TournamentPage/index.tsx:
```diff
@@ -0,0 +1,125 @@
+import { MatchNode } from './components/MatchNode';
+import * as styles from './styles.css';
+
+export type Player = {
```
`@/api/types` 내에 정의되어 있어요.

hakilee commented on src/pages/TournamentPage/index.tsx:
```diff
@@ -0,0 +1,125 @@
+  };
+
+  const handleBack = () => {
+    window.history.back();
```
`react-router-dom`의 `navigate(-1)` 를 사용해 주세요.

hakilee reviewed (commented):
고생하셨습니당 토너먼트 페이지가 Game의 하위에 존재하는데, 폴더는 pages에 있는 것 같아요. 확인 부탁드립니다~

yeni023 committed:
Merge branch 'dev' into feature/Tournament

yeni023 committed:
feat: Tournament 타입 정의 추가 및 Match 구조 설계

HJ0311 reviewed (approved):
수고하셨습니다!

hakilee reviewed (approved):
고생하셨습니다! 잘 구현된 것 같아요. 👍
